### PR TITLE
[#16417] Remove unnecessary format functionality

### PR DESCRIPTION
**Pull request by willjthorp**
Closes https://github.com/Leadfeeder/issue-tracker/issues/16417

It's not necessary to define a format for the value, as we can just JSON stringify / parse the value and it will work properly.

This PR removes the normalize from string util, and changes the logic so the tracked local storage class will stringify the value being set in browser local storage, then parse it when retrieving the value.

(It also adds the `keyNameProperty` feature which we are currently using in the monorepo package)

### How to test

You can fetch and run the PR locally, then grabbing the `tracked-local-storage` service from ember inspector, and try manually setting / getting some local storage props of different types, and make sure everything is working properly.

### CODE REVIEW

File: addon/utils/tracked-local-storage.js [hunk 1]
```diff
@@ -50,7 +50,15 @@ export default class TrackedLocalStorage {
...
-    return value === DOES_NOT_EXIST ? null : value;
+    if (value === DOES_NOT_EXIST) {
+      return null;
+    }
+
+    try {
+      return JSON.parse(value);
+    } catch (e) {
+      return value;
+    }
```
Comment (willjthorp) [inline addon/utils/tracked-local-storage.js:61]: Try catch needed for backwards compatibility for values which were previous stored as plain strings

**Comment by willjthorp** [comment]
Review @jbandura @tszymon

**Comment by willjthorp** [comment]
:white_check_mark: Review @jbandura @Darren-LF

**Review by jbandura — APPROVED** [review]
Code looks good 👍

**Review by Darren-LF — APPROVED** [review]

**Pull request merged**